config/locales/doc_auth/en.yml

```diff
@@ -12,7 +12,7 @@ en:
     forms:
       address1: Address
       change_file: Change file
-      choose_file: Drag file here or choose from folder
+      choose_file_html: Drag file here or <u>choose from folder</u>
```

Comment:
So I see that the `<u>` will be replaced client-side with a `<span>` -- what was the reasoning for picking `u`? Do we want the underline as a "fallback" style? I guess I'm asking why not just pick a `<span>` or something more similar anyways

Reply:
> Do we want the underline as a "fallback" style? I guess I'm asking why not just pick a `<span>` or something more similar anyways

Yeah, I contemplated a few approaches here:

Like you mention, we could anticipate a case where we might render this outside React and want some sensible fallback. `<u>` is the closest visual alternative (though I've now come to find that it [may not be an appropriate _semantic_ alternative](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/u)).

On the other hand, I considered if we should be very explicit that these markup strings are assumed to be used in an environment where the substitution would be applied (our new `formatHTML`), in which case the element could be named in a way to try to signal this. As referenced in a few of the example test cases and documentation, I was thinking the [native custom element syntax](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements) could work well for this. For example, something like `<lg-underline>`. It's also valid HTML, and won't _break_ if rendered elsewhere; it just won't have the associated underlined styling.

On the third (😄 ) hand, and as raised by @solipet , there's definitely still an option to just [dangerously set the inner HTML](https://reactjs.org/docs/dom-elements.html#dangerouslysetinnerhtml), and use whatever markup here in this file as we'd want to be rendered regardless of React or Rails. It might be fair to say that if it's safe enough to render unescaped in Rails, it's safe enough to render unescaped in React? But I tend toward avoiding `innerHTML` at all costs 😬

Reply:
I like the idea of `<lg-underline>` and we just map it to something styled the way we want in JS

Reply:
> I like the idea of `<lg-underline>` and we just map it to something styled the way we want in JS

Updated in fb01c7c.